src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/AsyncCompletionLogger.cs

```diff
@@ -10,11 +10,16 @@ internal static class AsyncCompletionLogger
     {
         private static readonly LogAggregator s_logAggregator = new();
         private static readonly StatisticLogAggregator s_statisticLogAggregator = new();
+        private static readonly HistogramLogAggregator s_histogramLogAggregator = new(25, 500);
 
         private enum ActionInfo
         {
-            // For type import completion
+            // For import completion
             SessionWithTypeImportCompletionEnabled,
+            SessionWithImportCompletionBlocking,
```

Comment:
`SessionWithImportCompletionBlocking` records the number of sessions where import completion is enabled and we wait for them to complete (because the editor's responsive completion option is disabled). Subtracting it from `SessionWithTypeImportCompletionEnabled` gives us the number of sessions the delay could happen (if import completion is slow.)

You are correct about the other two :)